queue.go

```diff
@@ -165,6 +166,7 @@ func (q *Queue) Put(val interface{}, timeout float64) error {
 	q.clearPending()
 	isfull := q.isfull()
 	if timeout < 0.0 && isfull {
+		q.mutex.Unlock()
```

Comment:
`fmt.Println("Test concurrent nowait Put/Get...if deadlock block forever or panic!")`
`go func() {
		defer wg.Done()
		for {
			select {
			case <-putDone:
				return
			default:
				_, _ = queue.GetNoWait()
			}
		}
	}()
	go func() {
		defer wg.Done()
		time.Sleep(3 * time.Second)
		for i := 0; i < 100; i++ {
			_ = queue.PutNoWait(1)
		}
		putDone <- true
	}()`
Concurrent run the two goroutines if not add the unlock operation the process may be block and deadlock or panic....